### ISSUE TITLE
UX: Help with migration from old versions

### ISSUE THREAD

**Issue by L-TChen** [agda/agda#4325]
One often finds that each release of Agda breaks existing their codebase, although the best effort has been made to prevent such regressions. Sometimes, it is even expected. This is however a longstanding issue, and many Agda users I know suffer from it.

I think that it might be a good idea to include some instructions to help people spot and migrate their code to the new version. A number of possible solutions I can think of are

* Migration guide — like GHC which provides a migration guid for each major release

* CI script template for type checking with the latest commit for each Agda repository— this can be useful to gather more bug reports.

* ??? 

[ I put this into icebox since it is unlikely to solve easily ...]

**Comment by asr** [comment]
Are you talking about in the second item of something similar to [this](https://github.com/agda/agda/issues/4239#issuecomment-561375671) comment? If so, I agree (but I don't see its relation with 'migration from old versions').

**Comment by L-TChen** [comment]
Yes, I think all we really need is a brief migration guide indicating regressions and changes which may cause any problem.

Anything else would be useful not only for migration but also development in general.

**Comment by andreasabel** [comment]
I think we have now adopted the practice to mark breaking changes as such in the changelog.
Is there something actionable left for this issue, @L-TChen ?

**Comment by L-TChen** [comment]
I couldn't think of anything actionable for now. This issue matters less now as migration can be done with LLMs. And also I just feel it is okay not to make code updated, as many Agda projects are research-oriented POCs. 😅
